README.md

```diff
@@ -1,24 +1,211 @@
-# README
+# Next-Gen-Auth
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+A basic multi-page rails demo app created to showcase [Shopify's Next-Gen
+Auth][1] using [Turbolinks][2]. This demo shows how to convert multi-page apps
+into Next-Gen Auth enabled apps.
 
-Things you may want to cover:
+If you have a regular multi-page app and you want to use the new Session Token
+Auth but you're unsure or not yet ready to convert your app to an SPA, you can
+still use Session Tokens by converting your app to use Turbolinks. Turbolinks
+is a JavaScript library that makes your application behave as if it were a
+single-paged application.
 
-* Ruby version
+## Getting started
 
-* System dependencies
+> __Important:__ This guide assumes your app has the Next-Gen Auth beta feature
+> enabled. This provides your app early access to use JWT based authorization.
+> Reach out to someone on the App Authorization team if you are unsure whether your
+> app has this enabled.
 
-* Configuration
+If you are looking to install this app on your own testing environment, and
+already have a `Shopify API Key` and `Shopify API Secret` from your partners
+dashboard, you can pull this repository and run the following:
 
-* Database creation
+```console
+$ bundle install
+$ yarn install
+$ rails db:migrate
+```
+See [this section][3] for a guide on setting up your `.env` file.
 
-* Database initialization
+You will also need to use [ngrok][4] to make `localhost:3000` available to the
+internet. You should use the secure HTTPS URL generated by ngrok to create your
+app with on your Partners dashboard.
```

Comment:
Should we add more details to this section? Is it well-known that they need to use the ngrok URL and then use URL/auth/shopify/callback when setting this up in the dashboard? Or could we add a link for them to refer to how to do this step just in case it's already documented but they might have forgotten how to do it?

Reply:
This guide has a section on using Ngrok https://www.shopify.in/partners/blog/shopify-admin-authenticate-app

Reply:
I'm going to swap out link [4] with Adi's link. I think it would be more useful than to link to ngrok's home page.